static/app.js

```diff
@@ -42,6 +41,46 @@ $(function() {
         console.log('Geolocation not enabled in this browser')
     }
 
+    let timeoutId;
+
+    $('#term').on('keyup', ()=> {
+        if ($('#search_type').val() == 'by keyword') {
+            clearTimeout(timeoutId);
+            timeoutId = setTimeout(runAutocomplete, 500)
```

Comment:
Typically debouncing is used here. This is something you can explore with an external library if necessary

It's also a good opportunity to explore Debounce vs Throttle

Reply:
new code here...

/** set up event handler for keyup on input field, checks whether search type is by keyword
     * calls debounced function runAutoComplete
     */

    $('#term').on('keyup', ()=> {
        if ($('#search_type').val() == 'by keyword') {
            debouncedRunAutoComplete()
        } 
    });

    /** debounce function that sets up debouncing on a passed in function */

    function debounce(func, delay = 500) {
        let timeoutId;
        return (...args) => {
            clearTimeout(timeoutId)
            timeoutId = setTimeout(()=> {
                func(...args)
            }, delay)
        }
    }

    /** assignment of the debounced function, in this case runAutoComplete  */

    const debouncedRunAutoComplete = debounce(runAutoComplete)